src/core/runner.ts

```diff
@@ -233,10 +238,26 @@ export default class Runner extends EventEmitter {
     driver.context.on('request', captureUrl);
 
     try {
+      /**
+       * Set up plugin maanger context and also register
+       * step level plugins
+       */
+      const traceEnabled = trace || filmstrips;
       pluginManager.onStep(step);
+      traceEnabled && (await pluginManager.start('trace'));
+      // call the step definition
       await step.callback();
+      /**
+       * Collect all step level metrics
+       */
       if (metrics) {
-        data.metrics = await pluginManager.get(PerformanceManager).getMetrics();
+        data.pagemetrics = await (
+          pluginManager.get('performance') as PerformanceManager
+        ).getMetrics();
+      }
+      if (traceEnabled) {
+        const traceOutput = await pluginManager.stop('trace');
+        Object.assign(data, { ...traceOutput });
```

Comment:
What is the difference between this and `Object.assign(data, traceOutput);` ? They are the same no?

Reply:
Yes fixed it.